### PR TITLE
DOC: Note on trailing commas and indexing for R users

### PR DESCRIPTION
**Pull request by nickeubank**
Related to https://discourse.julialang.org/t/dataframes-why-is-df-2-the-same-as-df-2/6053

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

**Comment by nalimilan** [comment]
Thanks! I have made a few edits, since data frames are not part of the Base language, so better give an example about matrices instead (and just because an object is called `df`, it's not obvious it's a data frame). Hopefully we'll improve the behavior of data frames at some point so that they are more consistent with matrices.

**Review by StefanKarpinski — APPROVED** [review]

**Comment by KristofferC** [comment]
`Be aware that Julia ignores trailing commas.`

Is this really true though?
y

File: doc/src/manual/noteworthy-differences.md [hunk 1]
```diff
@@ -107,6 +107,10 @@ For users coming to Julia from R, these are some noteworthy differences:
   * Like many languages, Julia does not always allow operations on vectors of different lengths, unlike
     R where the vectors only need to share a common index range.  For example, `c(1, 2, 3, 4) + c(1, 2)`
     is valid R but the equivalent `[1, 2, 3, 4] + [1, 2]` will throw an error in Julia.
+  * Be aware that Julia ignores trailing commas. This can cause confusion among R users when indexing
```
Comment (KristofferC) [inline doc/src/manual/noteworthy-differences.md]: > Be aware that Julia ignores trailing commas.

Isn't this a bit too general of a statement?

 ```
julia> (1)
1

julia> (1,)
(1,)

 julia> (1,,)
 ERROR: syntax: unexpected ,
```

Comment (nalimilan) [reply doc/src/manual/noteworthy-differences.md]: Maybe add "when indexing"?

Comment (nickeubank) [reply doc/src/manual/noteworthy-differences.md]: My understanding from the discussion on discourse is that:
   - Julia will ignore ONE (and only one) trailing comma, UNLESS
   - that comma is used to signal something is a tuple

I guess I can flesh that whole nuance out? But it's not limited to indexing. e.g. 

    julia> [1, 2, 3,]
    3-element Array{Int64,1}:
     1
     2
     3

Comment (nickeubank) [reply doc/src/manual/noteworthy-differences.md]: julia> [1, 2, 3,,]
    ERROR: syntax: unexpected ,

**Comment by StefanKarpinski** [comment]
I think a better phrase would be "Julia allows an optional trailing comma without changing meaning".

**Comment by nickeubank** [comment]
That look alright?

File: doc/src/manual/noteworthy-differences.md [hunk 2]
```diff
@@ -107,6 +107,10 @@ For users coming to Julia from R, these are some noteworthy differences:
   * Like many languages, Julia does not always allow operations on vectors of different lengths, unlike
     R where the vectors only need to share a common index range.  For example, `c(1, 2, 3, 4) + c(1, 2)`
     is valid R but the equivalent `[1, 2, 3, 4] + [1, 2]` will throw an error in Julia.
+  * Julia allows an optional trailing comma when that comma does not change the meaning of code.
+   This can cause confusion among R users when indexing into arrays. For example, `x[1,]` in R
```
Comment (StefanKarpinski) [inline doc/src/manual/noteworthy-differences.md]: Not sure if this affects rendering, but this is indented one space less than one would expect.

**Review by nalimilan — APPROVED** [review]

File: doc/src/manual/noteworthy-differences.md (hunk 2 above)
Comment (nalimilan) [reply doc/src/manual/noteworthy-differences.md]: I fixed it directly.

**Comment by StefanKarpinski** [comment]
Thanks for the clarification, @nickeubank!

**Pull request merged**